web/src/unread_ops.ts

```diff
@@ -256,31 +258,71 @@ export function mark_as_unread_from_here(
     }
     message_lists.current.prevent_reading();
 
-    // If we have already fully fetched the current view, we can
-    // send the server the set of IDs to update, rather than
-    // updating on the basis of the narrow.
-    let message_ids_to_update;
-    if (message_lists.current.data.fetch_status.has_found_newest()) {
-        message_ids_to_update = message_lists.current
-            .all_messages()
-            .filter(
-                (msg) =>
-                    (include_anchor && msg.id >= message_id) ||
-                    (!include_anchor && msg.id > message_id),
-            )
-            .map((msg) => msg.id);
+    const rounding_factor = 25;
+    const threshold_fetched = 50;
+    const threshold_unfetched = 10;
+    const is_all_fetched = message_lists.current.data.fetch_status.has_found_newest();
+    const is_in_interleaved_view = !narrow_state.narrowed_by_reply();
```

Comment:
This check is incorrect; `filter.is_conversation_view() || filter.is_conversation_view_with_near()` is closer to what you want. (near/with views are not accounted for correctly, which have extra operators whose only impact is on the scroll position, not which messages match).

But I guess actually even that check would be wrong: A narrow that has `sender/channel/topic` filters would not be interleaved, because it's a subset of a given conversation.

So maybe you want `current_filter.can_bucket_by("channel", "topic") || filter.can_bucket_by("dm"))` as the check here; that seems like it precisely asks if this contains a single conversation.

Probably reasonable to make that value available as an exported function `filter.is_interleaved_view` in a prep commit, since it took me a while to figure out how to define it.